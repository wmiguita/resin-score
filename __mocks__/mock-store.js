import configureStore from "redux-mock-store"

const mockStore = configureStore([])

//TODO: find better way to keep reducers up to date and check for any test side effects
export const store = mockStore({
  atheletes: {
  },
  userInterface: {
    drawerOpen: false,
    selectedTab: 0
  }
})
