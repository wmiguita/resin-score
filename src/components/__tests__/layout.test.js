import React from "react"
import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import { wrapComponent } from "mock-store"

import Layout from "../layout"

const pageProps = { location: { pathname: "/" } }
let layout

describe( "Layout", () => {
  it( "renders Layout with children", () => {
    const tree = renderer.create( wrapComponent (
      <Layout { ...pageProps }><div>child</div></Layout>
    )).toJSON()

    expect( tree ).toMatchSnapshot()
  })

  it( "renders Layout with props", () => {
    const propedLayout = renderer.create( wrapComponent(
      <Layout { ...pageProps }><div></div></Layout>
    )).toJSON()

    expect( propedLayout ).toMatchSnapshot()
  })
})
