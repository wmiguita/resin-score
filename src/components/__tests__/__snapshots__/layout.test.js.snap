// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders Layout with children 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6"
      >
        
      </h1>
    </div>
  </header>,
  <main
    className="content"
  >
    <div
      className="makeStyles-appBarSpacer-3"
    />
    <div>
      child
    </div>
  </main>,
]
`;

exports[`Layout renders Layout with props 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6"
      >
        
      </h1>
    </div>
  </header>,
  <main
    className="content"
  >
    <div
      className="makeStyles-appBarSpacer-3"
    />
    <div />
  </main>,
]
`;
